apps/rush-lib/src/cli/utilities/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -123,19 +123,53 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
     // version will be either:
     // A - the version (e.g. "0.0.5")
     // B - a peer dep version (e.g. "/gulp-karma/0.0.5/karma@0.13.22"
+    //                           or "/@ms/sp-client-utilities/3.1.1/foo@13.1.0"
     //                           or "/sinon-chai/2.8.0/chai@3.5.0+sinon@1.17.7")
 
     // check to see if this is the special style of specifiers
-    // e.g.:  "/gulp-karma/0.0.5/karma@0.13.22"
-    // split it by forward slashes, then grab the second group
+    // e.g.:  "/gulp-karma/0.0.5/karma@0.13.22" or
+    //     or "/@ms/sp-client-utilities/3.1.1/foo@13.1.0"
+    // split it by forward slashes, then grab the second group (or the 3rd, if the package name is scoped)
     // if the second group doesn't exist, return the version directly
     if (version) {
-      const versionParts: string[] = version.split('/');
-      if (versionParts.length !== 1 && versionParts.length !== 4) {
+      try {
+        const versionParts: string[] = version.split('/');
+
+        // it had no slashes, so we know it is a version like "0.0.5"
+        if (versionParts.length === 1) {
+          return version;
+        }
+
+        // this should never happen
+        if (versionParts.length === 2 || versionParts.length === 3) {
+          throw undefined;
+        }
+
+        const isScoped: boolean = versionParts[1].indexOf('@') === 0;
+
+        // if it has 4 parts, then it should be unscoped
+        if (versionParts.length === 4) {
+          if (isScoped) {
+            throw undefined;
```

Comment:
>throw [](start = 12, length = 5)

according to our coding conventions, nobody should ever throw any object type of any kind under any circumstances unless it inherits from the Error class